src/azure-cli/azure/cli/command_modules/aro/_aad.py

```diff
@@ -3,109 +3,55 @@
 
-import datetime
 import time
-import uuid
 
-from azure.cli.core._profile import Profile
-from azure.cli.core.commands.client_factory import configure_common_settings
-from azure.cli.core.azclierror import BadRequestError
-from azure.graphrbac import GraphRbacManagementClient
-from azure.graphrbac.models import ApplicationCreateParameters
-from azure.graphrbac.models import GraphErrorException
-from azure.graphrbac.models import PasswordCredential
-from azure.graphrbac.models import ServicePrincipalCreateParameters
+from azure.cli.command_modules.role import graph_client_factory
+from azure.cli.command_modules.role import GraphError
+
 from knack.log import get_logger
 
 logger = get_logger(__name__)
 
 
+# All methods now return the text based values
+# for consistency and hiding implementation details
 class AADManager:
     def __init__(self, cli_ctx):
-        profile = Profile(cli_ctx=cli_ctx)
-        credentials, _, tenant_id = profile.get_login_credentials(
-            resource=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
-        self.client = GraphRbacManagementClient(
-            credentials, tenant_id, base_url=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
-        configure_common_settings(cli_ctx, self.client)
+        self.client = graph_client_factory(cli_ctx)
 
     def create_application(self, display_name):
-        password = uuid.uuid4()
-        start_date = datetime.datetime.utcnow()
-        end_date = datetime.datetime(2299, 12, 31, tzinfo=datetime.timezone.utc)
-
-        app = self.client.applications.create(ApplicationCreateParameters(
-            display_name=display_name,
-            identifier_uris=[],
-            password_credentials=[
-                PasswordCredential(
-                    custom_key_identifier=str(start_date).encode(),
-                    start_date=start_date,
-                    end_date=end_date,
-                    value=password,
-                ),
-            ],
-        ))
-
-        return app, password
+        request_body = {"displayname": display_name, }
+        app = self.client.application_create(request_body)
+        obj_id = app["id"]
+        app_id = app["appId"]
+        password = self.add_password(obj_id)
+        return app_id, password
 
     def get_service_principal(self, app_id):
-        sps = list(self.client.service_principals.list(
-            filter="appId eq '%s'" % app_id))
+        sps = self.client.service_principal_list(f"appId eq '{app_id}'")
         if sps:
-            return sps[0]
+            return sps[0]["id"]
         return None
 
     def get_application_by_client_id(self, client_id):
-        apps = list(self.client.applications.list(
-            filter="appId eq '%s'" % client_id))
+        apps = self.client.application_list(f"appId eq '{client_id}'")
         if apps:
-            return apps[0]
+            return apps[0]["id"]
```

Comment:
Perhaps the method should be renamed to `get_application_id_by_client_id` to better reflect what it is doing?

Reply:
Pushed a new commit to incorporate this.